### PR TITLE
tiltfile: k8s_service takes yaml as an OPTIONAL arg [ch715]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @nicks, @jazzdan,

Please review the following commits I made in branch maiamcc/yaml-opt-arg:

671c8cda19dbf0dc3d8cb718a863e2b9771edf89 (2018-11-01 18:00:01 -0400)
tiltfile: k8s_service takes yaml as an OPTIONAL arg

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,12 +146,20 @@ class Service
     """
 ```
 
+#### global_yaml
+Call this _on the top level of your Tiltfile_ on a string of YAML.
```
Comment (jazzdan) [inline README.md]:
```suggestion
Call this _on the top level of your Tiltfile_ with a string of YAML.
```

File: README.md [hunk 2]
```diff
@@ -146,12 +146,20 @@ class Service
...
+
+We will infer what (if any) of the k8s resources defined in your YAML correspond to `Services` defined elsewhere in your Tiltfile (matching based on the DockerImage ref and on pod selectors). Any remaining YAML is _global YAML_, i.e. YAML that Tilt applies to your k8s cluster independently of any `Service` you define. 
+```python
+def global_yaml(yaml: string) -> None
+```
+
 #### k8s_service
 
-Creates a kubernetes service that tilt can deploy using the yaml text and the image passed in.
+Creates a kubernetes service that tilt can deploy using the the image passed in. Optionally, you may also pass the Kubernetes resource YAML. If the YAML is not passed, we expect to be able to extract it from `global_yaml` (see above).
```
Comment (jazzdan) [inline README.md]: Will we error here if we can't find it?

File: internal/tiltfile/tiltfile.go [hunk 3]
```diff
@@ -24,6 +24,9 @@ import (
 
 const FileName = "Tiltfile"
 
+const oldK8sServiceSyntaxError = "he syntax for `k8s_service` has changed. Before it was `k8s_service(yaml: string, dockerImage: Image)`. " +
```
Comment (jazzdan) [inline internal/tiltfile/tiltfile.go]:
```suggestion
const oldK8sServiceSyntaxError = "The syntax for `k8s_service` has changed. Before it was `k8s_service(yaml: string, dockerImage: Image)`. " +
```

File: README.md (hunk 2 above)
Comment (maiamcc) [reply README.md]: good Q. i believe we won't error right here, but later on, when we call manifest.Validate()

**Comment by nicks** [comment]
do you know what the error looks like if you build an image, but it's not in the local or global yaml? i forget if we have good tests for that case.

**Comment by nicks** [comment]
also, if this gets committed today, i'd love to do the 0.1 release!

**Comment by maiamcc** [comment]
@nicks if we get a manifest with no yaml, that error gets surfaced when we call `manifest.Validate()` in whatever BuildAndDeployer we're using -- which isn't covered in tests anywhere, i don't think, cuz pretty sure we don't have tests for the individual BADs?

**Pull request merged**